### DISCUSSION TITLE
Integration with table libraries

### DISCUSSION THREAD

**Discussion by aaronschwartz** [edmundhung/conform#303] [Ideas]
I'm having trouble organizing my code to make it easy to render inputs using the `react-table` library.

Right now I have to call `useFieldSet` on the entire row for every table cell since I can't call the hook inside of a map to build the data upfront.
Current column definition:
```
const InputCell = ({formRef, row}) => {
    const {description} = useFieldset(formRef, row.original);
    return <input {...conform.input(description)} readOnly={true}/>
}
// column definition:
{
        accessorKey: "description",
        cell: ({row, table}) => {
            return <InputCell formRef={table.options.meta.formRef} row={row}/>;
        }
}
```

I wish there was a way to build up an array of the final data needed for the table. Maybe there could be a `subProperties` key or something with access to the full tree. 
```
// It would be nice to be able to do the equivalent of this (but this isn't possible since it's a hook)
const todos = useFieldList(form.ref, tasks);
const tableData = todos.map(todo=>useFieldset(form.ref, todo);

// Or have everything fully available from the start
const [form, { tasks }] = useForm({});
const tableData = tasks.todos.map(todo=>useFieldset(form.ref, todo);
```

Is there a better way to do this?

**Comment by rlesniak** [comment]
Same here. `Fieldset` idea is so... wrong 😞 disqualifies this lib for me
